File: src/com/l2kt/gameserver/network/clientpackets/RequestFriendList.java
package com.l2kt.gameserver.network.clientpackets;

import com.l2kt.gameserver.data.sql.PlayerInfoTable;
import com.l2kt.gameserver.model.World;
import com.l2kt.gameserver.model.actor.instance.Player;
import com.l2kt.gameserver.network.SystemMessageId;
import com.l2kt.gameserver.network.serverpackets.SystemMessage;

public final class RequestFriendList extends L2GameClientPacket
{
	@Override
	protected void readImpl()
	{
	}
	
	@Override
	protected void runImpl()
	{
		final Player activeChar = getClient().getActiveChar();
		if (activeChar == null)
			return;
		
		// ======<Friend List>======
		activeChar.sendPacket(SystemMessageId.FRIEND_LIST_HEADER);
		
		for (int id : activeChar.getFriendList())
		{
			final String friendName = PlayerInfoTable.getInstance().getPlayerName(id);
			if (friendName == null)
				continue;
			
			final Player friend = World.getInstance().getPlayer(id);
			
			activeChar.sendPacket(SystemMessage.getSystemMessage((friend == null || !friend.isOnline()) ? SystemMessageId.S1_OFFLINE : SystemMessageId.S1_ONLINE).addString(friendName));
		}
		
		// =========================
		activeChar.sendPacket(SystemMessageId.FRIEND_LIST_FOOTER);
	}
}

File: src/com/l2kt/gameserver/network/clientpackets/RequestFriendList.kt
package com.l2kt.gameserver.network.clientpackets

import com.l2kt.gameserver.data.sql.PlayerInfoTable
import com.l2kt.gameserver.model.World
import com.l2kt.gameserver.network.SystemMessageId
import com.l2kt.gameserver.network.serverpackets.SystemMessage

class RequestFriendList : L2GameClientPacket() {
    override fun readImpl() {}

    override fun runImpl() {
        val activeChar = client.activeChar ?: return

        activeChar.sendPacket(SystemMessageId.FRIEND_LIST_HEADER)

        for (id in activeChar.friendList) {
            val friendName = PlayerInfoTable.getInstance().getPlayerName(id) ?: continue

            val friend = World.getInstance().getPlayer(id)

            activeChar.sendPacket(
                SystemMessage.getSystemMessage(if (friend == null || !friend.isOnline) SystemMessageId.S1_OFFLINE else SystemMessageId.S1_ONLINE).addString(
                    friendName
                )
            )
        }
        activeChar.sendPacket(SystemMessageId.FRIEND_LIST_FOOTER)
    }
}